src/Juvix/Compiler/Pipeline.hs

```diff
@@ -40,43 +40,43 @@ compile ::
 compile = typecheck >=> C.fromInternal
 
 upToParsing ::
-  Members '[Reader EntryPoint, Files, Error JuvixError, NameIdGen] r =>
+  Members '[Embed IO, Reader EntryPoint, Files, Error JuvixError, NameIdGen] r =>
```

Comment:
Maybe we could add `UpdateStdlib` to the `Files` effect, similarly to what you did for `RegisterStdlib`. This way we don't need to have `Embed IO` in the pipeline

Reply:
good idea!